### PR TITLE
Do We Really Need Recompose? A Thought Experiment

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Please start the pull request title with the jira ticket corresponding to the work if applicable. Please include a short summary of the feature added, the change, or issue fixed.

## Type of Change
- Bug fix ('fix', 'repair', 'bug')
- Non breaking change ('update', 'change')
- Breaking change ('break', 'deprecate')

If the any above types of change apply to this pull request, please ensure to include one of the listed keywords in the pull request title.

## Applicable E2E Tests

To run relevant E2E tests, run these commands in 3 separate terminals:

1. `yarn && yarn start`
2. `yarn selenium`
3. `yarn e2e --spec=COMMA_SEPARATED_LIST_OF_SPECS_HERE --browser=headlessChrome`

## Note to Reviewers

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

### CODE REVIEW

File: packages/manager/src/features/Backups/types.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Linode, LinodeType } from 'linode-js-sdk/lib/linodes';
+import { BackupError } from 'src/store/backupDrawer';
+
+export interface LinodeWithTypeInfo extends Linode {
+  typeInfo?: LinodeType;
+}
+
+export interface ExtendedLinode extends LinodeWithTypeInfo {
+  linodeError?: BackupError;
+}
```
Comment (acourdavault) [inline packages/manager/src/features/Backups/types.ts]: should this be in a different PR?

File: packages/manager/src/utilities/compose.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export const compose = <TInner, TOuter>(
+  ...funcs: Function[]
+): ComponentEnhancer<TInner, TOuter> =>
+  funcs.reduce(
+    (a, b) => (...args: any) => a(b(...args)),
+    (arg: any) => arg
+  ) as ComponentEnhancer<TInner, TOuter>;
+
+export default compose;
```
Comment (acourdavault) [inline packages/manager/src/utilities/compose.ts]: We need a unit test for this, do they have one we could also import @johnwcallahan ?

File: packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/IPSharingPanel.tsx [hunk 3]
```diff
@@ -23,6 +22,7 @@ import TextField from 'src/components/TextField';
 import withLinodes, {
   DispatchProps
 } from 'src/containers/withLinodes.container';
+import { compose as recompose } from 'src/utilities/compose';
```
Comment (acourdavault) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/IPSharingPanel.tsx]: could we take this chance to not name this recompose as we are getting rid of the recompose library this naming is more confusing

**Comment by Jskobos** [comment]
Action window for this has closed. Will revisit later.
